sphinx_gherkindoc/writer.py

```diff
@@ -187,7 +188,11 @@ def format_step(step: behave.model.Step, step_format: str) -> str:
         # Make bold any scenario outline variables
         formatted_step = re.sub(r"(\\\<.*?\>)", r"**\1**", rst_escape(step.name))
         # Apply the step format string
-        formatted_step = step_format.format(f"{step.keyword} {formatted_step}")
+        # Every step keyword has the `step-keyword` role applied to it
+        # so that users can customize how the step keyword is formatted with CSS.
+        formatted_step = step_format.format(
+            f":step-keyword:`{step.keyword}` {formatted_step}"
```

Comment:
To clarify the previous comment about links being made from roles, when I looked at the roles documentation, it seemed to indicate that using a leading `!` would suppress any linkage for the roles, but having looked at the domains and plugins briefly, I wonder how that is controlled and if it is only needed for a role that would otherwise create a link.

As to nesting, I was looking at that from the point of view of having a role for the step keyword as is here, and how that might interact with a role for the entire step, etc.  It might be that a role for the entire step could inject the needed markup for assigning the css roles to the various parts (such as the keyword and possibly any text that represents variable substituion)?

Reply:
I can try to look into it. From that doc it looks like custom domains would need to be put into https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#more-domains, though maybe you can load one in locally.

I think that if collisions are the concern, which I think is valid, a little prefix makes perfect sense. Especially since we care less about the rST domain/role and more about the class that is given to the HTML element. I'll start to look ahead to the other places we could give roles. If they are all gherkin related, perhaps `gherkin-` would be good. `gherkin-step-keyword`, `gherkin-step-text`, etc.

Reply:
I also looked into having nested roles, which it appears is not possible in rST. So unless we find another way to do it, we can only wrap independent pieces. It should be OK for now, though it would be nice to be able to apply formatting to say `step` for all the text, and then something special just for `step-keyword`. But again I think it is ok for now.

Reply:
I've added some more places to format in a WIP branch: https://github.com/rbcasperson/sphinx_gherkindoc/tree/more-formatting-customization

Doing that, it is clear that everything that we want to format is Gherkin related, so I like the idea of using `gherkin` as the prefix, especially since it will give better context in a CSS file that might also have non-gherkin formatting. So I'll go with that prefix here and in the future PRs as well, then further review can confirm or reject that.